third_party/zap/templates/helper-chip.js

```diff
@@ -0,0 +1,93 @@
+/*
+*
+*    Copyright (c) 2020 Project CHIP Authors
+*
+*    Licensed under the Apache License, Version 2.0 (the "License");
+*    you may not use this file except in compliance with the License.
+*    You may obtain a copy of the License at
+*
+*        http://www.apache.org/licenses/LICENSE-2.0
+*
+*    Unless required by applicable law or agreed to in writing, software
+*    distributed under the License is distributed on an "AS IS" BASIS,
+*    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*    See the License for the specific language governing permissions and
+*    limitations under the License.
+*/
+
+/**
+ * This module contains the API for templating. For more detailed instructions, read {@tutorial template-tutorial}
+ *
+ * @module Templating API: toplevel utility helpers
+ */
+
+/**
+ * Produces the top-of-the-file header for a C file.
+ *
+ * @returns The header content
+ */
+function chip_header() {
+    return `
+  /*
+  *
+  *    Copyright (c) 2020 Project CHIP Authors
+  *
+  *    Licensed under the Apache License, Version 2.0 (the "License");
+  *    you may not use this file except in compliance with the License.
+  *    You may obtain a copy of the License at
+  *
+  *        http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *    Unless required by applicable law or agreed to in writing, software
+  *    distributed under the License is distributed on an "AS IS" BASIS,
+  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *    See the License for the specific language governing permissions and
+  *    limitations under the License.
+  */`;
+}
+
+function isClient(side) {
+    return 0 == side.localeCompare("client");
+}
+
+function isServer(side) {
+    return 0 == side.localeCompare("server");
+}
+
+function isStrEqual(str1, str2) {
+    return 0 == str1.localeCompare(str2);
+}
+
+function isLastElement(index, count) {
+    return index == count - 1;
+}
+
+function isEnable(enable) {
+    return 1 == enable;
+}
+
+function isCommandAvailable(clusterSide, incoming, outgoing, source, name) {
+    if (0 == clusterSide.localeCompare(source)) {
+        return false;
+    }
+
+    if (isClient(clusterSide) && outgoing) {
+        return true;
+    } else if (isServer(clusterSide) && incoming) {
+        return true;
+    }
+    return false;
```

Comment:
nit: That's just preference. But since those if blocks just return true if the are called I would use:

```
if (0 == clusterSide.localeCompare(source)) {
    return false;
}

if (isClient(clusterSide) && outgoing) {
    return true;
}

if (isServer(clusterSide) && incoming) {
    return true;
}

return false;
```

But feel free to ignore.

Reply:
Difference of opinion here, I prefer having checks that are related inside a `if {} else if`